src/newgidmap.c

```diff
@@ -60,7 +61,7 @@ static bool verify_range(struct passwd *pw, struct map_range *range, bool *allow
 	}
 
 	/* Allow a process to map its own gid. */
-	if ((range->count == 1) && (pw->pw_gid == range->lower)) {
+	if ((range->count == 1) && (getgid() == range->lower)) {
```

Comment:
Interesting.  If user is using non-primary id, should they still be allowed to
map pw->pw_gid?  Probably not, so this is probably right, and this may be a
bugfix.